### PR TITLE
Email editor - Document usage of private APIs from WP [MAILPOET-6319]

### PR DESCRIPTION
**Pull request by costasovo**
## Description

I searched for all unlock usages and moved them into a single file, `private-apis/index.ts` and removed the unlock export so that we can't use it in any other place. The goal is to have all private APIs we use in one place and with comments. I found a couple of cases where we could get rid of the unlock call and I removed one unnecessary private hook. 

## Code review notes

The very first commit is not related to the private APIs. To work with the newest version of Gutenberg, you need to set `template-locked` mode. I ran into that because I was testing with the new Gutenberg, so I fixed it immediately. I hope it is ok to include it here without an additional ticket.

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-6319]

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-6319]: https://mailpoet.atlassian.net/browse/MAILPOET-6319?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:email-editor/private-apis)

_The latest successful build from `email-editor/private-apis` will be used. If none is available, the link won't work._

### CODE REVIEW

File: packages/js/email-editor/src/private-apis/index.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	// @ts-expect-error No types for privateApis.
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
+import { privateApis as componentsPrivateApis } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+
+const { unlock } = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+	'I acknowledge private features are not for use in themes or plugins and doing so will break in the next version of WordPress.',
+	'@wordpress/edit-post' // The module name must be in the list of allowed, so for now I used the package name of the post editor
+);
```
Comment (triple0t) [inline packages/js/email-editor/src/private-apis/index.ts:15]: I noticed you switched from using the import [here](https://github.com/mailpoet/mailpoet/blob/e728e74d1166a5e572e90aee105a829d4ba0bc30/packages/js/email-editor/src/private-apis/lock-unlock.js). Instead you redefined the same variable. May I ask why that is so? Won't it be better to have a single unlock interface?

File: packages/js/email-editor/src/private-apis/index.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/**
+ * WordPress dependencies
+ */
+import { __dangerousOptInToUnstableAPIsOnlyForCoreModules } from '@wordpress/private-apis';
+import {
+	// @ts-expect-error No types for privateApis.
+	privateApis as blockEditorPrivateApi,
+	// @ts-expect-error No types for privateApis.
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
```
Comment (triple0t) [inline packages/js/email-editor/src/private-apis/index.ts]: There is a single `privateApis` [export](https://github.com/WordPress/gutenberg/blob/9b0e2d96679ab6c82033946bead9521529e6b29a/packages/block-editor/src/index.js#L23). You can simplify this section.

**Review by triple0t — COMMENTED** [review]
LGTM. Thank you for working on this.

I have minor remarks; kindly locate them below.

File: packages/js/email-editor/src/private-apis/index.ts (hunk 1 above)
Comment (costasovo) [reply packages/js/email-editor/src/private-apis/index.ts:15]: Yes. This was intentional. I removed the other one in [a later commit](https://github.com/mailpoet/mailpoet/pull/5948/commits/e427d17cf3ba4df6ec79e862de07bee078ff5a5c). This unlock is not exported. This is for us to "prevent" adding new private API usage directly and "force" us to do all unlocking in this place. Does that make sense?

File: packages/js/email-editor/src/private-apis/index.ts (hunk 2 above)
Comment (costasovo) [reply packages/js/email-editor/src/private-apis/index.ts]: Nice catch.  This is not intentional.

File: packages/js/email-editor/src/private-apis/index.ts (hunk 1 above)
Comment (triple0t) [reply packages/js/email-editor/src/private-apis/index.ts:15]: In that case, can you delete this file https://github.com/mailpoet/mailpoet/blob/e427d17cf3ba4df6ec79e862de07bee078ff5a5c/packages/js/email-editor/src/private-apis/lock-unlock.js

Comment (costasovo) [reply packages/js/email-editor/src/private-apis/index.ts:15]: Aha. I see I have 2 now 🤦

Comment (costasovo) [reply packages/js/email-editor/src/private-apis/index.ts:15]: Thank you for noticing this. I missed that one.

File: packages/js/email-editor/src/private-apis/index.ts (hunk 2 above)
Comment (triple0t) [reply packages/js/email-editor/src/private-apis/index.ts]: You need to update this too https://github.com/mailpoet/mailpoet/blob/1bf95f4b0a554e275006b24940e5a7e21a7cae31/packages/js/email-editor/src/private-apis/index.ts#L62

from `blockEditorPrivateApi`
to `blockEditorPrivateApis`

**Comment by costasovo** [comment]
@triple0t I fixed the issues and it should be ready for second review.

**Review by triple0t — APPROVED** [review]
LGTM
Thank you

**Pull request merged**